File: codes/web/src/types/api/views/analytics.ts
export type MyAnalyticsView = {
  display_name: string;
  email: string;
  task_total: number;
  todo_count: number;
  doing_count: number;
  done_count: number;
  archived_count: number;
  high_priority_count: number;
  health_percent: number | null;
  workspace_count: number;
  project_count: number;
};

File: codes/core-service/app/schemas/views/analytics.py
from pydantic import BaseModel


class MyAnalyticsViewOut(BaseModel):
    display_name: str
    email: str
    task_total: int
    todo_count: int
    doing_count: int
    done_count: int
    archived_count: int
    high_priority_count: int
    health_percent: int | None
    workspace_count: int
    project_count: int
